### PR TITLE
Extend General fee with Sanitary Dump Stations

### PR DESCRIPTION
**Pull request by catdogmat**
This quest applies to ~50% of all sanitary dump stations on OSM. I extended the General Fee Quest, and added Sanitary Dump Stations.

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/quests/general_fee/AddGeneralFee.kt [hunk 1]
```diff
@@ -12,7 +12,7 @@ class AddGeneralFee : OsmFilterQuestType<Boolean>() {
 
     override val elementFilter = """
         nodes, ways with
-         (tourism = museum or leisure = beach_resort or tourism = gallery or tourism = caravan_site)
+         (tourism = museum or leisure = beach_resort or tourism = gallery or tourism = caravan_site or amenity = sanitary_dump_station)
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/general_fee/AddGeneralFee.kt:15]:
```suggestion
         tourism ~ museum|gallery|caravan_site
         or leisure = beach_resort
         or amenity = sanitary_dump_station
```

**Comment by catdogmat** [comment]
I think this is ready to merge.

**Review by westnordost — CHANGES_REQUESTED** [review]
Hmm, but the question posed is "Do you need to pay to enter here?"

The wording doesn't really fit for sanitary dump stations. Can a wording be found that would fit all these places?

**Comment by catdogmat** [comment]
How about "Is there an entrance fee or cost associated with this location?"

**Comment by Helium314** [comment]
Couldn't there just be a separate title for sanitary dump stations, or maybe generalized to places / things one does not (want to) enter?

**Comment by westnordost** [comment]
Yes, that would be possible. I would also prefer such a solution. The new string could then say something like whether it costs something to use it.

**Comment by catdogmat** [comment]
Ok sounds good, closing then.

**Comment by westnordost** [comment]
Hu, why? I think there has been a misunderstanding. The PR is fine, but we'd need a second question text.

Notice how the `getTitle` function in line 26 has the tags as a parameter? You could check whether it is a sanitary dump station and in that case instead return a different string, something like "Do you ahve to pay ... to **use**...?"

**Comment by qugebert** [comment]
@catdogmat this should work
   ```
 override fun getTitle(tags: Map<String, String>) =
        if (tags["amenity"] == "sanitary_dump_station")
            R.string.quest_generalFee_title
        else
            R.string.quest_generalFee_title2
```
and add a second title to app/src/main/res/values/strings.xml

**Comment by catdogmat** [comment]
Alright, I'll add this now

**Comment by catdogmat** [comment]
Thanks, @qugebert i've added it

**Comment by catdogmat** [comment]
Alright, @westnordost I think this is ready for a merge!

File: app/src/main/java/de/westnordost/streetcomplete/quests/general_fee/AddGeneralFee.kt [hunk 2]
```diff
@@ -12,7 +12,9 @@ class AddGeneralFee : OsmFilterQuestType<Boolean>() {
 
     override val elementFilter = """
         nodes, ways with
-         (tourism = museum or leisure = beach_resort or tourism = gallery or tourism = caravan_site)
+         tourism ~ museum|gallery|caravan_site
+         or leisure = beach_resort
+         or amenity = sanitary_dump_station
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/general_fee/AddGeneralFee.kt:17]:
```suggestion
         (
           tourism ~ museum|gallery|caravan_site
           or leisure = beach_resort
           or amenity = sanitary_dump_station
         )
```

missing brackets

**Review by westnordost — CHANGES_REQUESTED** [review]

**Comment by catdogmat** [comment]
Ok, @westnordost I fixed the brackets

**Review by westnordost — APPROVED** [review]
Thanks, looks fine now!

**Comment by catdogmat** [comment]
I think this is ready to merge, @westnordost

**Comment by catdogmat** [comment]
Oh wait does @FloEdelmann need to review first

**Review by FloEdelmann — APPROVED** [review]
I'm fine with the changes, too :slightly_smiling_face:

**Comment by westnordost** [comment]
It will be merged for the next main release (not for a bugfix release, which I expect the next release to be)

**Pull request merged**